src/likelihoods/likelihoods.jl

```diff
@@ -0,0 +1,10 @@
+"""
+    Likelihood
+
+An abstract type for likelihoods. Likelihoods are used to model the uncertainities associated
+with the data.
+"""
+abstract type Likelihood end
```

Comment:
Is the abstract type needed and part of the design? I'm not sure if it's actually useful since no methods are defined for it and it's not part of any type restrictions (and I guess we don't want these restrictions to allow for any kinds of likelihoods).

Reply:
It might be useful while defining links which @theogf suggested earlier in [this comment](https://github.com/JuliaGaussianProcesses/LatentGPs.jl/pull/3#issuecomment-634599211).

Reply:
How will it help when defining links?

Reply:
This depends on how we design Links. One possible way while making use of `Bijectors.jl` would be
```julia
struct Link
    lik::Likelihood
    transform::Bijector
end
(link::Link)(x) = link.lik(link.transform(x))

#while making a LatentGP
link = Link(PoissonLikelihood(), Exp())
lgp = LatentGP(gp, link)
```
We could maybe come up with something more self-explanatory than "Link".

Reply:
But that's not actually an argument in favour of a `Likelihood` type, you probably just want to have
```julia
struct Link{L,B<:Bijector}
    lik::L
    transform::B
end
```
to allow all kinds of user-defined likelihood functions. `Likelihood` is not used anywhere else in the example, it seems.

Reply:
Yeah you are right. I will remove Likelihood type. Thanks!